lua/completion-treesitter/source.lua

```diff
@@ -0,0 +1,91 @@
+local ts_locals = require'nvim-treesitter.locals'
+local parsers = require'nvim-treesitter.parsers'
+local ts_utils = require'nvim-treesitter.ts_utils'
+
+local M = {}
+
+M.match_kinds = {
+  var = 'v',
+  method = 'm',
+  ['function'] = 'f'
+}
+
+local function get_smallest_context(source)
+  -- Step 1 get current context
+  local scopes = ts_locals.get_scopes()
+  local current = source
+
+  while current ~= nil and not vim.tbl_contains(scopes, current) do
+    current = current:parent()
+  end
+
+  return current or nil
+end
+
+local function prepare_match(match, kind)
+  local matches = {}
+
+  if match.node then
+    table.insert(matches, { kind = M.match_kinds[kind] or '', full = match.node, def = match.node })
+  else
+    for name, item in pairs(match) do
+      vim.list_extend(matches, prepare_match(item, name))
+    end
+  end
+
+  return matches 
+end
+
+function M.getCompletionItems(prefix, score_func, bufnr)
+  if parsers.has_parser() then
+    local at_point = ts_utils.get_node_at_cursor()
+    local _, line_current, _, _, _ = unpack(vim.fn.getcurpos())
+
+    local complete_items = {}
+
+    -- Step 2 find correct completions
+    for _, definitions in ipairs(ts_locals.get_definitions(bufnr)) do
+      local matches = prepare_match(definitions)
+
+      for _, match in ipairs(matches) do
+        local node = match.def
+        local node_scope = get_smallest_context(node)
+        local start_line_node, _, _= node:start()
+
+        local node_text = ts_utils.get_node_text(node, bufnr)[1]
+
+        -- Only consider items in current scope, and not already met
+        local score = score_func(prefix, node_text)
+
+        if score < #prefix / 2 and node_text
```

Comment:
Is the filtering still handled by us ? I have not followed completion-nvim dev recently but did it changed ?

Reply:
I'll double check.